File: client/app/src/main/java/com/qubacy/geoqq/domain/mate/message/model/MateMessage.kt
package com.qubacy.geoqq.domain.mate.message.model

import com.qubacy.geoqq.domain._common.model.user.User

data class MateMessage(
    val id: Long,
    val user: User,
    val text: String,
    val time: Long
) {
}

File: server/own_services/common/pkg/storage/geoqq/sql/postgre/mateMessage.go
package postgre

import (
	"common/pkg/domain/geoqq"
	"common/pkg/postgreUtils/wrappedPgxpool"
	utl "common/pkg/utility"
)

func MateMessageWithChatIdFromQueryResult(scanner wrappedPgxpool.QueryResultScanner) (
	*geoqq.MateMessageWithChat, error) {
	sourceFunc := MateMessageWithChatIdFromQueryResult

	mm := geoqq.MateMessageWithChat{}
	err := scanner.Scan(
		&mm.Id, &mm.ChatId,
		&mm.UserId, &mm.Text,
		&mm.Time, &mm.Read)
	if err != nil {
		return nil, utl.NewFuncError(sourceFunc, err)
	}

	return &mm, nil // ok!
}
